CMakeLists.txt

```diff
@@ -17,6 +17,22 @@ endif()
 
 project(az LANGUAGES C)
 enable_testing ()
-add_subdirectory(sdk/core/core)
-add_subdirectory(sdk/keyvault/keyvault)
-# add_subdirectory(sdk/storage/ustorageclient)
+
+# make options for each subdirectoy so it can be selected what to build on CI Pipelines
+option(BUILD_AZ_CORE "Build lib and targets from Azure core" ON)
+option(BUILD_AZ_KEY_VAULT_CLIENT "Build lib and targets from Azure Key Vault client" OFF)
+option(BUILD_AZ_USTORAGE_CLIENT "Build targets from u-storage client" OFF)
+
+if (BUILD_AZ_CORE)
+  add_subdirectory(sdk/core/core)
+endif()
+if (BUILD_AZ_KEY_VAULT_CLIENT)
```

Comment:
All of these conditional builds are very complicated and error prone. Could we keep it simple, at least for now, for example build everything?